js/modules/k6/experimental/csv/module.go

```diff
@@ -0,0 +1,286 @@
+	}
+
+	// Create a new Parser instance
+	parser := Parser{
+		reader:  r,
+		options: options,
+		vu:      mi.vu,
+	}
+
+	return rt.ToValue(&parser).ToObject(rt)
+}
+
+// Next returns the next row in the CSV file.
+func (p *Parser) Next() *sobek.Promise {
+	promise, resolve, reject := promises.New(p.vu)
+
+	go func() {
+		var records []string
+		var done bool
+		var err error
+
+		records, err = p.reader.Read()
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				done = true
+			} else {
+				reject(err)
+				return
+			}
+		}
+
+		p.currentLine.Add(1)
+
+		resolve(parseResult{Done: done, Value: records})
+	}()
+
+	return promise
+}
+
+// parseResult holds the result of a CSV parser's parsing operation such
+// as when calling the [Next].
+type parseResult struct {
+	// Done indicates whether the parser has finished reading the file.
+	Done bool `js:"done"`
+
+	// Value holds the line's records value.
+	Value []string `js:"value"`
+}
+
+// options holds options used to configure CSV parsing when utilizing the module.
+//
+// The options can be used to either configure the CSV parser, or the parse function.
+// They offer to customize the behavior of the parser, such as the delimiter, whether
+// to skip the first line, or  to start reading from a specific line, and stop reading
+// at a specific line.
+type options struct {
+	// Delimiter is the character that separates the fields in the CSV.
+	Delimiter rune `js:"delimiter"`
+
+	// SkipFirstLine indicates whether the first line should be skipped.
+	SkipFirstLine bool `js:"skipFirstLine"`
+
+	// FromLine indicates the line from which to start reading the CSV file.
+	FromLine null.Int `js:"fromLine"`
+
+	// ToLine indicates the line at which to stop reading the CSV file (inclusive).
+	ToLine null.Int `js:"toLine"`
+}
+
+// newDefaultParserOptions creates a new options instance with default values.
+func newDefaultParserOptions() options {
+	return options{
+		Delimiter:     ',',
+		SkipFirstLine: false,
+	}
+}
+
+// newParserOptions creates a new options instance from the given Sobek object.
+func newParserOptionsFrom(obj *sobek.Object) (options, error) {
+	options := newDefaultParserOptions()
+
+	if obj == nil {
+		return options, nil
+	}
+
+	if v := obj.Get("delimiter"); v != nil {
+		delimiter := v.String()
+
+		// A delimiter is gonna be treated as a rune in the Go code, so we need to make sure it's a single character.
+		if len(delimiter) > 1 {
+			return options, fmt.Errorf("delimiter must be a single character")
+		}
+
+		options.Delimiter = rune(delimiter[0])
+	}
+
+	if v := obj.Get("skipFirstLine"); v != nil {
+		options.SkipFirstLine = v.ToBoolean()
+	}
+
+	if v := obj.Get("fromLine"); v != nil {
+		options.FromLine = null.IntFrom(v.ToInteger())
+	}
+
+	if v := obj.Get("toLine"); v != nil {
+		options.ToLine = null.IntFrom(v.ToInteger())
+	}
+
+	if options.FromLine.Valid && options.ToLine.Valid && options.FromLine.Int64 > options.ToLine.Int64 {
```

Comment:
This doesn't seem to match the equivalent in `csv/reader.go:66`, as there it seems to be checking `>=` while here it just checks `>`.

Personally, I think this is the correct one, because as `ToLine` is inclusive, if you just want to parse one line, you would set `FromLine` and `ToLine` to the line you want to read, but either this one or the other needs to be adjusted. I leave it up to you!